activerecord/test/cases/suppressor_test.rb

```diff
@@ -0,0 +1,18 @@
+require 'cases/helper'
+require 'models/notification'
+require 'models/user'
+
+class SuppressorTest < ActiveRecord::TestCase
+
+  def test_executes_callbacks_without_suppression_setting
+    count = Notification.count
+    UserWithNotification.create!
+    assert_equal count + 1, Notification.count
+  end
+
+  def test_suppresses_creation_of_related_record_on_after_create_callback
+    count = Notification.count
+    Notification.suppress { UserWithNotification.create! }
+    assert_equal count, Notification.count
```

Comment:
Should assert that other models used in the suppress block -- in this case, User -- are unaffected:

```ruby
assert_difference -> { User.count } do
assert_no_difference -> { Notification.count } do
  Notification.suppress { UserWithNotification.create! }
end
end
```